org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/contentmodel/participants/diagnostics/XMLValidator.java

```diff
@@ -400,4 +408,22 @@ private static void updateExternalGrammarLocation(DOMDocument document, SAXParse
 			}
 		}
 	}
+
+	/**
+	 * Report error if the path refered to by the catalog URI is invalid
+	 * 
+	 */
+	private static void checkForInvalidCatalogURI(DOMDocument document, List<Diagnostic> diagnostics) {

```

Comment:
Dont update this file. The xml catalog validation must be done in the catalog extensions bycreationthe ckass XMLCatalogDiagnosticParticipant that you need to register in https://github.com/eclipse/lemminx/blob/fd0cec5f78db475d98893ebaf8dd101005a4f9b2/org.eclipse.lemminx/src/main/java/org/eclipse/lemminx/extensions/catalog/XMLCatalogPlugin.java#L64